components/NoteEditor/Editor/Editor.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import dynamic from 'next/dynamic';
+
+import shortid from 'shortid';
+import styles from './Editor.module.scss';
+import Heading from './Heading';
+
+const Section = dynamic(() => import('./Section'), { ssr: false });
+
+function formatDate(dateObj) {
+  return new Intl.DateTimeFormat('en-US', { month: 'short', day: 'numeric' }).format(dateObj);
+}
+
+const formatedDate = formatDate(new Date()).split(' ');
+
+export default function Editor() {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [sections, setSections] = useState([{ id: shortid.generate(), keyword: '', content: '' }]);
+  const [summary, setSummary] = useState('');
+
+  const handleAddSection = () => {
+    const newSection = { id: shortid.generate(), keyword: '', content: '' };
+    setSections(sections.concat(newSection));
+  };
+
+  const handleRemoveSection = (e) => {
+    const sectionId = e.currentTarget.dataset.target;
+
+    if (sections.length === 1) return;
+
+    if (sectionId) {
+      const index = sections.findIndex((section) => section.id === sectionId);
+      setSections([...sections.slice(0, index), ...sections.slice(index + 1, sections.length)]);
+    } else {
+      setSections(sections.slice(0, sections.length - 1));
+    }
+  };
+
+  const handleInput = (e) => {
+    const { id } = e.currentTarget.parentElement.dataset;
+    const key = (e.currentTarget.classList[0].match(/Editor_content.+/))
+      ? 'content' : 'keyword';
+
+    const updatedSections = sections.map((section) => {
+      if (section.id === id) {
+        const updated = {
+          ...section,
+          [key]: e.currentTarget.innerHTML,
+        };
+
+        return updated;
+      }
+      return section;
+    });
+
+    setSections(updatedSections);
+  };
+
+  const handleSummary = (e) => {
+    setSummary(e.currentTarget.innerHTML);
+  };
+
+  // eslint-disable-next-line arrow-body-style
+  const renderSections = () => sections.map((section) => {
+    return (
+      <Section
+        key={section.id}
+        id={section.id}
+        handleInput={handleInput}
+        handleRemove={handleRemoveSection}
+      />
+    );
+  });
+
+  return (
+    <div className={styles.editor}>
+      <header className={styles.header}>
+        <div className={`${styles.headings} ${styles.wordbreak}`}>
+          <Heading name="title" placeholder="Title" value={title} setValue={setTitle} />
+          <Heading name="description" placeholder="Note description" value={description} setValue={setDescription} />
```

Comment:
The component naming is a bit misleading imo, something like AdjustableTextarea would probably make more sense, or what do you think?

Reply:
:shrug: I'm not good with naming. Sounds good to me